### ISSUE TITLE
[question] Where are entry JS scripts referenced in HTML?

### ISSUE THREAD

**Issue by BrainCrumbz** [preboot/angular2-webpack#36]
Hello, trying to figure out how webpack bundling works with such a setup:

from webpack configuration it seems that `vendor.js` and `app.js` are the two bundles that should be built. Then (from our understanding) those should be referenced in some HTML page. But we could not find any reference to those. Could you help us figure out what are we missing?
Thanks

**Comment by Foxandxss** [comment]
Sure mate.

[Here](https://github.com/preboot/angular2-webpack/blob/master/webpack.config.js#L184-L206) will you find what you need.

Basically we get the `index.html` as template and then webpack will inject there all the <script> tags it needs and <link> tags (for CSS). That way you don't need to maintain a .html file that could potentially need 1 .js file or 20 (same for css). Also for development the bundle files names are different from the bundle files from production (this is not 100% sure right now but I will change that soon).

In short, we provide the basic html and we let webpack inject all the needed bits to work.

**Comment by BrainCrumbz** [comment]
I see.. so you're modifying the HTML not only to fill in some values (as we saw doing elsewhere), but also to actually add the needed script nodes.

**Comment by Foxandxss** [comment]
The only thing we insert in our templates are the script tags and link tags.

It is mostly to avoid some manual tasks. Every time you generate a new bundle, it would have a hash, so you would need to modify the html to put the new hash. That is a PITA so webpack will do that for you.

**Comment by BrainCrumbz** [comment]
Great. Thanks for clarifying! I guess we can close this (question) issue.
